src/components/uri/Visualization.tsx

```diff
@@ -598,9 +598,16 @@ function buildAfmResultSpec(
     sortingConfigs: ISortingConfigs,
 ): IConvertedAFM {
     const updatedVisObj = fillMissingTitles(visObj, locale);
-    const genericAfmResultSpec = toAfmResultSpec(updatedVisObj);
-
-    return buildAfmResultSpecForVis(updatedVisObj, visType, genericAfmResultSpec, sortingConfigs);
+    // Ignore the measures' titles to make sure afm response contains the up-to-date titles
+    const ignoredMeasureTitlesVisObj = ignoreTitles(updatedVisObj);
```

Comment:
why do we `fillMissingTitles` then `ignoreTitles`?

Reply:
Titles are ignored before creating an AFM request to make sure the AFM response to contain the latest titles; otherwise, the titles included in AFM request (may be outdated) are returned back as-is, causing the Visualization component/KD to display out-of-date titles.